### PR TITLE
boards: shields: Add Semtech SX1262MB2DAS board

### PR DESCRIPTION
**Pull request by papadkostas**
Adding support for Semtech SX1262 board.
Tested succefully with lora send and lora receive.

Signed-off-by: Konstantinos Papadopoulos <kostas.papadopulos@gmail.com>

### CODE REVIEW

File: boards/shields/semtech_sx1262mb2das/semtech_sx1262mb2das.overlay [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+		lora0 = &lora;
+	};
+};
+
+&arduino_spi {
+	status = "okay";
+
+	cs-gpios = <&arduino_header 13 GPIO_ACTIVE_LOW>;
+
+	lora: sx1262@0 {
```
Comment (erwango) [inline boards/shields/semtech_sx1262mb2das/semtech_sx1262mb2das.overlay:17]:
```suggestion
	lora_semtech_sx1262mb2das: sx1262@0 {
```

File: boards/shields/semtech_sx1262mb2das/semtech_sx1262mb2das.overlay [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022 Konstantinos Papadopoulos <kostas.papadopulos@gmail.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/ {
+	aliases {
+		lora0 = &lora;
```
Comment (erwango) [inline boards/shields/semtech_sx1262mb2das/semtech_sx1262mb2das.overlay:8]:
```suggestion
		lora0 = &lora_semtech_sx1262mb2das;
```

**Review by erwango — CHANGES_REQUESTED** [review]
node label to be renamed. See https://docs.zephyrproject.org/latest/hardware/porting/shields.html#shield-porting-and-configuration. This is made to avoid name conflicts

**Review by erwango — DISMISSED** [review]

File: boards/shields/semtech_sx1262mb2das/doc/index.rst [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+| D13                   | SX1262 SPI SCK  |
++-----------------------+-----------------+
+
+The SX1262 signals DIO2 and DIO3 are not available at the shield connector.
+
+Requirements
+************
+
+This shield can only be used with a board which provides a configuration
+for Arduino connectors and defines node aliases for SPI and GPIO interfaces
```
Comment (jfischer-no) [inline boards/shields/semtech_sx1262mb2das/doc/index.rst:45]: Please remove  "_and defines node aliases for SPI and GPIO interfaces_", this statement is not correct.

**Review by jfischer-no — COMMENTED** [review]
Please fix your commit message:
```

shields: add Semtech SX1262MB2DAS

The SX1262MB2DAS comes...
```

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Please fix your commit message:
```

shields: add Semtech SX1262MB2DAS

The SX1262MB2DAS comes...
```

**Review by jfischer-no — CHANGES_REQUESTED** [review]
Please change the commit message to be like:
```

shields: add Semtech SX1262MB2DAS

The SX1262MB2DAS comes...
```

**Review by jfischer-no — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**